test/lib/mentionsPlugin.test.js

```diff
@@ -39,6 +39,14 @@ describe('mentionsPlugin', () => {
       expect(mentionTokens.length).to.equal(0);
     });
 
+    it('does not parse mentions in e-mail addresses ending in punctuation', () => {
```

Comment:
@mbroshi this test is correctly passing but I tried with the example posted in the original issue (`admin@habitica.com-test)` and it's still being counted as a mention

Reply:
Thanks for this catch! You are totally correct. I just opened an issue on linkify-it: https://github.com/markdown-it/linkify-it/issues/93

Other punctuation seems to be fine, but not hyphens. I will wait to hear back from them.